internal/proto/replication.proto

```diff
@@ -220,3 +196,28 @@ message GetVolumeReplicationInfoResponse{
   // This field is OPTIONAL.
   int64 last_sync_bytes = 3;
 }
+
+// Specifies what source the replication will be created from. One of the
+// type fields MUST be specified.
+message ReplicationSource {
+  // VolumeSource contains the details about the volume to be replication
+  message VolumeSource {
+    // Contains identity information for the existing volume.
+    // This field is REQUIRED.
+    string volume_id = 1;
+  }
+  // VolumeGroupSource contains the details about
+  // the volume group to be replication
+  message VolumeGroupSource {
+    // Contains identity information for the existing volume group.
+    // This field is REQUIRED.
+    string volume_group_id = 1;
+  }
+
+  oneof type {
+    // Volume source type
+    VolumeSource volume = 1;
+    // Volume group source type
+    VolumeGroupSource volumegroup = 2;
+  }
+}
```

Comment:
```suggestion
message ReplicationSource {
    oneof source {
    // Volume source type
    VolumeSource volume = 1;
    // Volume group source type
    VolumeGroupSource volumegroup = 2;
   }
  }

// VolumeSource contains the details about the volume to be replication
  message VolumeSource {
    // Contains identity information for the existing volume.
    // This field is REQUIRED.
    string volume_id = 1;
  }
  // VolumeGroupSource contains the details about
  // the volume group to be replication
  message VolumeGroupSource {
    // Contains identity information for the existing volume group.
    // This field is REQUIRED.
    string volume_group_id = 1;
}
```



https://protobuf.dev/programming-guides/proto3/#oneof
https://www.tutorialspoint.com/protobuf/protobuf_compound_data_types.htm

oneOf type should be inside the message ?

Reply:
https://github.com/csi-addons/spec/tree/main/replication#replicationsource 
Now I see that its present in csiaddons spec too.

Or am i mistaken :thinking:

Reply:
please disregard this comment.
I was mistaken :sweat:  I didn't read the indentation properly

more examples in https://github.com/container-storage-interface/spec/blob/master/spec.md